core/Symbols.cc

```diff
@@ -722,13 +722,24 @@ MethodRef findConcreteMethodTransitiveInternal(const GlobalState &gs, ClassOrMod
     }
 
     MethodRef result = owner.data(gs)->findMethod(gs, name);
+    if (result.exists() && result.data(gs)->flags.isOverloaded) {
+        auto overloadName = gs.lookupNameUnique(UniqueNameKind::Overload, name, 1);
+        result = owner.data(gs)->findMethod(gs, overloadName);
+    }
+
```

Comment:
Why are we doing this, instead of handling overloads at the call site of `findConcreteMethodTransitiveInternal`?

Reply:
Because all the information about whether or not something is abstract is currently stored on the first sig, instead of on the `<U ...>` named symbol. One option could be that we propagate all that information from the overload sigs back to the original symbol, and only use the overloads to host the multiple signatures, which would mean we wouldn't need this case anymore.

Reply:
> Because all the information about whether or not something is abstract is currently stored on the first sig, instead of on the `<U ...>` named symbol

ah that makes sense... 

yeah idk I guess let's leave it like this and figure out what to do later. maybe leave a comment or an issue link that says something about this all being completely broken anyways.

Reply:
That sounds good, I think there are a bunch of nice cleanups we can stack on top of this :)